### DISCUSSION TITLE
无网或者外网差，如何使用add_requires加载本地包

### DISCUSSION THREAD

**Discussion by mengxinxiaoqiang** [xmake-io/xmake#4260] [Q&A] [answered]
![image](https://github.com/xmake-io/xmake/assets/56524631/651bec23-7393-409b-b3c0-ed528ad26377)
已经设置了--pkg_searchdirs和--network=private，Debug模式不会去网络下载，Release还是会的，我应该如何设置？

**Comment by waruqi** [comment]
你本地缓存清下试试，`rm -rf .xmake`

**Reply by mengxinxiaoqiang** [reply]
还是一样的，删掉本地缓存还是会从网络下载
![image](https://github.com/xmake-io/xmake/assets/56524631/5c74cf9a-df1e-4355-a946-8fd898e7fc40)
.zip是Debug寻找的包名，.7z是Releases寻找的，如果这两个都存在，那就不会去网络下载，但无论是Debug还是Releases都可以用.zip里的代码进行本地编译来使用，不用网络下载，或者这里有什么考量？
![image](https://github.com/xmake-io/xmake/assets/56524631/f88f1f93-3f4f-4a6b-a5d6-19b287f687f1)

**Reply by waruqi** [reply]
7z 那个是预编译包。。 尽管你设置了 private 不连外网，但是 你又配置了 add_requires ，那么必须要自己提供本地包源，让它能提前找到，否则还是得从外网拉，不然前后需求矛盾

预编译包是加速安装用的，优先会去使用，你只提供本地源码包源，避免不了预编译包的拉取，除非你自建预编译仓库，提供内网包源，优先走你内网的

或者配置 `set_policy("package.precompiled", false)` 强制禁用所有预编译包拉取

**Reply by mengxinxiaoqiang** [reply, marked as answer]
是的配置了set_policy("package.precompiled", false) 强制禁用所有预编译包拉取，就能全部走内网了，只要提供源码包就行
